lib/shopify_app/controller_concerns/sanitized_params.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module SanitizedParams
+    protected
+
+    def sanitized_shop_name
+      @sanitized_shop_name ||= sanitize_shop_param(params)
+    end
+
+    def referer_sanitized_shop_name
+      return unless request.referer.present?
+
+      @referer_sanitized_shop_name ||= begin
+        referer_uri = URI(request.referer)
+        query_params = Rack::Utils.parse_query(referer_uri.query)
+
+        sanitize_shop_param(query_params.with_indifferent_access)
+      end
+    end
+
+    def sanitize_shop_param(params)
+      return unless params[:shop].present?
+      ShopifyApp::Utils.sanitize_shop_domain(params[:shop])
```

Comment:
I wonder if this would be better placed in the library since it might be useful for non-rails apps too.

Reply:
Let's do that as a separate PR/refactoring